dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/RotatingDnsResolver.kt

```diff
@@ -44,12 +46,18 @@ internal class RotatingDnsResolver(
 
         return if (knownHost != null && isValid(knownHost)) {
             knownHost.rotate()
-            knownHost.addresses
+            safeCopy(knownHost.addresses)
         } else {
             @Suppress("UnsafeThirdPartyFunctionCall") // handled by caller
             val result = delegate.lookup(hostname)
             knownHosts[hostname] = ResolvedHost(hostname, result.toMutableList())
-            result
+            safeCopy(result)
+        }
+    }
+
+    private fun safeCopy(list: List<InetAddress>): List<InetAddress> {
```

Comment:
I asked myself the same question but for now I could not see any other place. My rule of thumb is that an extension function should have at least 2 different classes where is being used in order to be extracted so that's the reason why I decided to keep it private, non extension here.